### PR TITLE
Fixed #177 pagination for all types

### LINKED ISSUE #177: `get_all_droplets()` broken
The example from the readme:
```
import digitalocean
manager = digitalocean.Manager(token=TOKEN)
my_droplets = manager.get_all_droplets()
print(my_droplets)
```
Appears to be broken:
```
Traceback (most recent call last):
  File "test.py", line 5, in <module>
    my_droplets = manager.get_all_droplets()
  File "digitalocean/Manager.py", line 101, in get_all_droplets
    data = self.get_data("droplets/")
  File "digitalocean/Manager.py", line 45, in get_data
    return self.__deal_with_pagination(args[0], data, params)
  File "digitalocean/Manager.py", line 67, in __deal_with_pagination
    values.append(value)
AttributeError: 'dict' object has no attribute 'append'
```

Running version: `python-digitalocean==1.10.1`
On Python 3.5.2

It appears to have been broken in 1.10.0, it worked in 1.9.0

### PR DESCRIPTION
**Pull request by WoLpH**
This fixes #177 and enables pagination for all API types.

### CODE REVIEW

**Comment by koalalorenzo** [comment]
Any chance to update the tests properly?

**Comment by WoLpH** [comment]
Yes, I was already working on it. This change breaks many tests unfortunately but it's going well :)

**Comment by koalalorenzo** [comment]
I am going to review but consider: if you are then changing the test, there is something wrong at the base of the changes applied.

**Comment by WoLpH** [comment]
The premise of this change is that every `GET` API call supports the `per_page` parameter and could have pagination. Note that all `POST` calls also support the `per_page` parameter but silently ignore it.

The result is that every call will fetch up to 200 items straight away and continues fetching as long as there are more pages available. While I didn't run the tests before I created the pull request, I've run extensive integration tests by using it in our production system. Since that system has over a thousand servers I can guarantee you that it's working better than before.

**Comment by koalalorenzo** [comment]
It would be super cool to have tests for 200+ servers! 👍 well done 😸

**Comment by koalalorenzo** [comment]
@WoLpH news about this? :)

**Comment by WoLpH** [comment]
Mostly that I've caught the flu and am not able to do much right now. Hopefully I can finish it this weekend

**Comment by koalalorenzo** [comment]
If you can update the pull request maybe somebody can pick it up where you left :)

**Comment by WoLpH** [comment]
It's done actually. Git was bugging me about requiring merges which were already merged. I'm working on a new pull request but somehow my local git seems broken... "git push -f wolph master" does nothing even though it's behind...

**Comment by WoLpH** [comment]
Not a clue what happend with my local git repo but I manually had to remove the remote and the `FETCH_HEAD` before it worked again. The new version tests correctly again and included the multipage fetching for all get requests if supported :)

File: digitalocean/baseapi.py [hunk 1]
```diff
@@ -97,6 +97,32 @@ def __perform_request(self, url, type=GET, params=None):
 
         return requests_method(url, **kwargs)
 
+    def __deal_with_pagination(self, url, method, params, data):
+        """
+            Perform multiple calls in order to have a full list of elements
+            when the API are "paginated". (content list is divided in more
+            than one page)
+        """
```
Comment (koalalorenzo) [inline digitalocean/baseapi.py]: Why the consistency of the single quote comments is not kept here? Suggestion: revert the changes related to double-single quote, and just focus on the problem that has been fixed in this section of the code! It is hard to review!

File: digitalocean/Droplet.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def create_multiple(*args, **kwargs):
 
         droplets = []
 
-        data = api.get_data("droplets", type=POST, params=data)
+        data = api.get_data('droplets/', type=POST, params=data)
```
Comment (koalalorenzo) [inline digitalocean/Droplet.py]: Is there any specific reason to change all of the double quotes into single quotes? It is hard to review and unnecessary to try to uniform this style only in some places.

File: digitalocean/baseapi.py (hunk 1 above)
Comment (WoLpH) [inline digitalocean/baseapi.py]: The double quotes are consistent with the other functions in this file. Personally I think a standard for either single or double quotes should be set for the entire project and should be changed in all cases. The changes were mostly accidental, when modifying/checking the code I ran my formatter on the code which converts to single quotes.

Comment (koalalorenzo) [inline digitalocean/baseapi.py]: Correct, the lack of inconsistency is the confusing part for me. In other place it was changed to single quotes. Don't worry, it is not a big issue as long as we can change it 👍 I suggest to just commit the real changes needed. I would prefer double quote btw.

**Comment by WoLpH** [comment]
The quotes are fixed now, beyond that it's still the same.

**Pull request merged**